db.js

```diff
@@ -308,15 +343,44 @@ var loans = (function () {
       return loanData;
     },
 
-    updateLoanById: function (loan) {
+    insert: async function (loan) {
       var sql =
-        "UPDATE loans SET requester_id = ?, book_id = ?, owner_contact = ?, " +
-        "requester_contact = ?, request_date = ?, accept_date = ?, loan_start_date = ?, " +
-        "loan_end_date = ?, return_date = ?, status = ? " +
-        "WHERE id = ?";
-      var inserts = [
+        "INSERT INTO loans (requester_id, book_id, requester_contact, owner_contact, status) " +
+        "VALUES (?,?,?,?,?)";
+      var initialInserts = [
         loan.requester_id,
         loan.book_id,
+        loan.requester_contact,
+        " ",
+        "pending",
+      ];
+      sql = mysql.format(sql, initialInserts);
+
+      var getStatusQuery = `SELECT COUNT (*) as "count" FROM loans WHERE requester_id = ? AND book_id = ?`;
+      var secondaryInserts = [loan.requester_id, loan.book_id];
+      getStatusQuery = mysql.format(getStatusQuery, secondaryInserts);
+
+      let currentLoan = await pool
+        .query(getStatusQuery)
+        .then(([rows, fields]) => {
+          return rows;
+        });
+
+      console.log(currentLoan[0].count);
+
+      if (currentLoan[0].count >= 1) {
+        return null;
```

Comment:
You'll want to return a Promise from the db file always. The routes are expecting a promise to append .then statements onto. You can do `return Promise.reject()` here instead. It will call the next `.catch()` block in the promise chain.